desc/basis.py

```diff
@@ -1413,83 +1412,150 @@ def zernike_radial_poly(r, l, m, dr=0, exact="auto"):
     return polyval_vec(coeffs, r, prec=prec).T
 
 
-@functools.partial(jit, static_argnums=3)
+@custom_jvp
+@jit
 def zernike_radial(r, l, m, dr=0):
     """Radial part of zernike polynomials.
 
-    Evaluates basis functions using JAX and a stable
-    evaluation scheme based on jacobi polynomials and
-    binomial coefficients. Generally faster for L>24
-    and differentiable, but slower for low resolution.
+    Calculates Radial part of Zernike Polynomials using Jacobi recursion relation
+    by getting rid of the redundant calculations for appropriate modes.
+    https://en.wikipedia.org/wiki/Jacobi_polynomials#Recurrence_relations
+
+    For the derivatives, the following formula is used with above recursion relation,
+    https://en.wikipedia.org/wiki/Jacobi_polynomials#Derivatives
+
+    Used formulas are also in the zerike_eval.ipynb notebook in docs.
+
+    This function can be made faster. However, JAX reverse mode AD causes problems.
```

Comment:
Should make an issue for this instead of including as a docstring, or put as a TODO comment somewhere where the problems occur

Reply:
#862 and #881 are related to this PR